ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/QuestionAnsweringModel.java

```diff
@@ -47,50 +63,230 @@ public void warmUp(Predictor predictor, String modelId, MLModelConfig modelConfi
+            return;
+        }
+
+        List<Input> overflowInputs = createOverflowInputs(question, context, overflowEncodings.length);
+        processOverflowInputs(overflowInputs, allHighlights);
+    }
+
+    private List<Input> createOverflowInputs(String question, String context, int numOverflowChunks) {
+        List<Input> overflowInputs = new ArrayList<>();
+        for (int i = 0; i < numOverflowChunks; i++) {
+            Input chunkInput = new Input();
+            chunkInput.add(MLInput.QUESTION_FIELD, question);
+            chunkInput.add(MLInput.CONTEXT_FIELD, context);
+            chunkInput.add(HIGHLIGHTING_MODEL_CHUNK_NUMBER_KEY, String.valueOf(i + 1));
+
+            overflowInputs.add(chunkInput);
+        }
+        return overflowInputs;
+    }
+
+    private void processOverflowInputs(List<Input> overflowInputs, List<Map<String, Object>> allHighlights) throws TranslateException {
+        try {
+            processOverflowInputsBatch(overflowInputs, allHighlights);
+        } catch (IllegalArgumentException e) {
+            log.info("Batch processing of chunks failed. Processing chunks individually: {}", e.getMessage());
+            processOverflowInputsIndividually(overflowInputs, allHighlights);
+        }
+    }
+
+    private void processOverflowInputsBatch(List<Input> overflowInputs, List<Map<String, Object>> allHighlights) throws TranslateException {
+        List<Output> overflowOutputs = getPredictor().batchPredict(overflowInputs);
+        for (Output output : overflowOutputs) {
+            try {
+                ModelTensors tensors = parseModelTensorOutput(output, null);
+                allHighlights.addAll(extractHighlights(tensors));
+            } catch (Exception e) {
+                log.warn("Error processing output from chunk", e);
+            }
+        }
+    }
+
+    private void processOverflowInputsIndividually(List<Input> overflowInputs, List<Map<String, Object>> allHighlights)
+        throws TranslateException {
+        for (int i = 0; i < overflowInputs.size(); i++) {
+            processOverflowChunkWithBatch(i + 1, overflowInputs.get(i), allHighlights);
+        }
+    }
+
+    /**
+     * Process a single overflow chunk using batchPredict and add any extracted highlights
+     * 
+     * @param chunkIndex The index of the overflow chunk (1-based)
+     * @param chunkInput The prepared input for this chunk
+     * @param highlights Collection to add extracted highlights to
+     */
+    private void processOverflowChunkWithBatch(int chunkIndex, Input chunkInput, List<Map<String, Object>> highlights)
+        throws TranslateException {
+        try {
+            // Use batchPredict instead of predict to avoid the bug
+            List<Output> outputs = getPredictor().batchPredict(List.of(chunkInput));
+            if (outputs.isEmpty()) {
+                log.warn("No output returned for chunk {}", chunkIndex);
```

Comment:
what would happen as following if the outputs is empty? does it worth proceeding forward?

Reply:
Updated to early return if the outputs is empty.